### PR TITLE
Adding concurrent request tests.

### PR DESCRIPTION
**Pull request by kyle-long**

### CODE REVIEW

File: spec/artifact-availability.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     * @return {Promise.<undefined>}
+     */
+    function createUploadDownloadChain() {
+        var count, promise;
+
+        count = 1;
+
+        promise = runUploadAndDownload();
+        for (count; count < 5; count += 1) {
```
Comment (GoogilyBoogily) [inline spec/artifact-availability.spec.js]: Why not set `count` inside the for loop like so `for (count = 1; count < 5; count += 1)`? If you want to do it this way, then you don't even need `count` in there. Like so `for (; count < 5; count += 1)`.

File: spec/artifact-availability.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            space = testSpace;
+        });
+    });
+    it("should be available in parallel", () => {
+        var count, promiseList;
+
+        count = 0;
+        promiseList = [];
+
+        for (count; count < config.concurrentRequests; count += 1) {
```
Comment (GoogilyBoogily) [inline spec/artifact-availability.spec.js]: Same comment as before.

File: spec/concurrency.spec.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ * @return {Promise.<Array.<Object>>}
+ */
+function runConcurrent(createPromise, dataList) {
+    var count, promise, promiseList;
+
+    count = 0;
+    promiseList = [];
+    dataList = dataList || [];
+
+    for (count; count < config.concurrentRequests; count += 1) {
```
Comment (GoogilyBoogily) [inline spec/concurrency.spec.js]: Same comment as before. Will stop commenting on them now, if there are more.

File: spec/concurrency.spec.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        if (dataList.length) {
+            promise = createPromise(dataList[count]);
+        } else {
+            promise = createPromise();
+        }
+        promiseList.push(promise);
```
Comment (GoogilyBoogily) [inline spec/concurrency.spec.js]: Need a newline above this.

File: spec/concurrency.spec.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            return runConcurrent((artifact) => {
+                logger.info(`Uploading a metadata property for ${artifact.uri}`);
+
+                return artifact.metadata.updateProperty(metadatum.name, metadatum);
+            }, space.artifactList);
+        }).then(() => {
+            return runConcurrent((artifact) => {
+                var criteria, search;
+
+                criteria = {
```
Comment (GoogilyBoogily) [inline spec/concurrency.spec.js:78]: Fun fact: if there is only one criteria, then it's called "criterion".

**Review by GoogilyBoogily — CHANGES_REQUESTED** [review]

File: spec/concurrency.spec.js (hunk 5 above)
Comment (kyle-long) [reply spec/concurrency.spec.js:78]: That is fun

File: spec/artifact-availability.spec.js (hunk 1 above)
Comment (kyle-long) [reply spec/artifact-availability.spec.js]: Could have sworn it was an eslint error. It doesn't appear to be though. Fixed

File: spec/artifact-availability.spec.js (hunk 2 above)
Comment (kyle-long) [reply spec/artifact-availability.spec.js]: same

File: spec/concurrency.spec.js (hunk 3 above)
Comment (kyle-long) [reply spec/concurrency.spec.js]: same. I changed all of them that I found

File: spec/concurrency.spec.js (hunk 4 above)
Comment (kyle-long) [reply spec/concurrency.spec.js]: Fixed

**Review by GoogilyBoogily — APPROVED** [review]
Looks great!

**Pull request merged**